### PR TITLE
Add cloud layer function for creating/deleting CCE cluster

### PR DESCRIPTION
**Pull request by gtema**

### CODE REVIEW

File: otcextensions/osclient/cce/v2/cluster.py [hunk 1]
```diff
@@ -97,18 +97,42 @@ def get_parser(self, prog_name):
...
+        ]:
+            if getattr(parsed_args, attr):
+                attrs[attr] = getattr(parsed_args, attr)
+
+        if not parsed_args.wait:
+            attrs['wait'] = False
+
         if parsed_args.cluster:
-            client = self.app.client_manager.cce
-            cluster = client.find_cluster(parsed_args.cluster,
```
Comment (tischrei) [inline otcextensions/osclient/cce/v2/cluster.py:106]: You have deleted find_cluster. Please change: "ID of the cluster" to "Name or ID of the cluster." in line 98 as well to let the user know that name is also possible.

File: otcextensions/sdk/cloud/cce.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+class CceMixin:
+
+    def create_cce_cluster(
+        self,
+        wait=True, wait_timeout=1800, wait_interval=5,
+        **kwargs
+    ):
+        """Create CCE cluster
+
+        :param str name: Cluster name.
```
Comment (tischrei) [inline otcextensions/sdk/cloud/cce.py:23]: Please use alphabetical order.

**Review by tischrei — CHANGES_REQUESTED** [review]

File: otcextensions/osclient/cce/v2/cluster.py [hunk 3]
```diff
@@ -97,18 +97,42 @@ def get_parser(self, prog_name):
             metavar='<cluster>',
             help=_('ID of the cluster')
         )
+        parser.add_argument(
+            '--wait',
+            action='store_true',
+            help=('Wait for the instance to become active')
```
Comment (lego963) [inline otcextensions/osclient/cce/v2/cluster.py]: did you miss here `_` ?

File: otcextensions/osclient/cce/v2/cluster.py [hunk 4]
```diff
@@ -181,47 +204,35 @@ def get_parser(self, prog_name):
...
-                wait_args['interval'] = parsed_args.wait_interval
-
-            client.wait_for_job(obj.job_id, **wait_args)
-            obj = client.get_cluster(obj.id)
+        for attr in [
+            'name', 'type', 'flavor', 'router', 'network',
+            'container_network_mode', 'annotations', 'authentication_proxy_ca',
+            'authentication_mode', 'container_network_cide',
+            'cpu_manager_policy', 'dss_master_volumes', 'description',
+            'external_ip', 'fix_pool_mask', 'labels', 'service_ip_range',
```
Comment (lego963) [inline otcextensions/osclient/cce/v2/cluster.py:222]: what do you think about renaming `labels` to `k8slabels`?

File: otcextensions/osclient/rds/v3/instance.py [hunk 5]
```diff
@@ -13,6 +13,7 @@
 '''Instance v1 action implementations'''
```
Comment (lego963) [inline otcextensions/osclient/rds/v3/instance.py]: Instance v1 -> Instance v3

**Review by lego963 — CHANGES_REQUESTED** [review]

File: otcextensions/osclient/cce/v2/cluster.py (hunk 4 above)
Comment (gtema) [reply otcextensions/osclient/cce/v2/cluster.py:222]: it's all belonging to k8. Not sure it make any sense. Generally here all names are pretty much how they are called in API with some normalizations.

File: otcextensions/sdk/cloud/cce.py (hunk 2 above)
Comment (gtema) [reply otcextensions/sdk/cloud/cce.py:23]: nope, can't be done

File: otcextensions/osclient/cce/v2/cluster.py (hunk 1 above)
Comment (gtema) [reply otcextensions/osclient/cce/v2/cluster.py:106]: done

**Review by lego963 — APPROVED** [review]

**Review by tischrei — APPROVED** [review]

**Pull request merged**